server/interaction/npc.go

```diff
@@ -0,0 +1,120 @@
+package interaction
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+const (
+	Dumb   model.PlayerID = `DumbNPC`
+	Simple model.PlayerID = `SimpleNPC`
+	Calc   model.PlayerID = `CalculatedNPC`
+)
+
+var npcs = map[model.PlayerID]npcLogic{
+	Dumb:   &dumbNPCLogic{},
+	Simple: &simpleNPCLogic{},
+	Calc:   &calcNPCLogic{},
+}
+
+var _ Player = (*NPCPlayer)(nil)
+
+type NPCPlayer struct {
+	HandleActionCallback func(ctx context.Context, a model.PlayerAction) error
+	logic                npcLogic
+	id                   model.PlayerID
+}
+
+// NewNPCPlayer creates a new NPC with specified type
+func NewNPCPlayer(pID model.PlayerID, cb func(ctx context.Context, a model.PlayerAction) error) (Player, error) {
+	l, ok := npcs[pID]
+	if !ok {
+		return &NPCPlayer{}, errors.New(`not a valid npc mode`)
```

Comment:
🌵 we shouldn't return "valid data" on error case, and this error could be something we expose publicly:

```
var (
  ErrUnknownNPCType error = errors.New(`unknown npc type`)
)
// ...

if !ok {
  return nil, ErrUnknownNPCType
}
```